File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 5/29/2026, 12:01:13 PM
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxw=0
        i=0
        j=len(height)-1
        while(i<j):
            maxw=max(maxw,min(height[i],height[j])*(j-i))
            if(height[i]<height[j]):
                i+=1
            else:
                j-=1
        return maxw

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.java
// Last updated: 5/29/2026, 12:01:19 PM
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int max = 0;
        while (left < right) {
            int width = right - left;
            int h = Math.min(height[left], height[right]);
            int water = h * width;
            max = Math.max(max, water);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return max;
    }
}
